src/populateExchange.ts

```diff
@@ -206,24 +201,32 @@ export const addFragmentsToQuery = ({
   userFragments,
 }: AddFragmentsToQuery) => {
   const typeInfo = new TypeInfo(schema);
-  const requiredUserFragments = new Set<FragmentDefinitionNode>();
-  let additionalFragments: Record<string, FragmentDefinitionNode> = {};
+
+  const requiredUserFragments: Record<
+    string,
+    FragmentDefinitionNode
+  > = Object.create(null);
+
+  const additionalFragments: Record<
+    string,
+    FragmentDefinitionNode
+  > = Object.create(null);
```

Comment:
I'm really thinking about when it gets more productive to make a `function createNullBaseObject` to avoid these 3 keywords

Reply:
that's pretty true. It'd be quickly done with a search & replace across the codebase. I'd actually just call it `createRecord`... Maybe for a separate where we can measure the impact? I'd expect gzip to take care of this though